File: backend/app/routers/notifications.py
from fastapi import APIRouter, Depends, status
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.dependencies import get_current_user
from app.models.user import User
from app.schemas.notification import MarkReadRequest, NotificationResponse
from app.services import notifications as notifications_service

router = APIRouter(prefix="/api/notifications", tags=["notifications"])


@router.get("", response_model=list[NotificationResponse])
async def list_notifications(
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    return await notifications_service.list_notifications(db, current_user.id)


@router.post("/read")
async def mark_read(
    body: MarkReadRequest,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    updated = await notifications_service.mark_read(db, current_user.id, body.ids, body.all)
    return {"updated": updated}

File: frontend/src/shared/api/notifications.ts
import http from './http'
import type { Notification } from './types'

export async function listNotifications(): Promise<Notification[]> {
  const { data } = await http.get<Notification[]>('/notifications')
  return data
}

export async function markRead(opts: { ids?: string[]; all?: boolean }): Promise<{ updated: number }> {
  const { data } = await http.post<{ updated: number }>('/notifications/read', opts)
  return data
}
